.github/workflows/main.yml

```diff
@@ -0,0 +1,23 @@
+name: Python package
+on:
+  push:
+    tags:
+      - '[0-9]+\.[0-9]+\.[0-9]+'
```

Comment:
Any reason you didn't use something like this rather than the `check-branch` step below?

```
    branches:
      - release
```

Reply:
This is where the fun comes in... It is good to keep in mind a ` tag` is not related to a `branch` at all. A `tag` lives on a specific commit which could be present on one or more branches. 

So with the github workflows the `on` command will look at a `branch` or a `tag` but it works like an `or` statement as far as I can tell. There is no way to `and` them. 

Due to this I could capture the `tag` in the specified format which would trigger the process and then introspect the branch that was currently pushed to and I *think* it will be able to pick it up. 

It would be much easier with a `v<x>.<y>.<z>` system which we could use independent of the branch and provide a bit more structure to the regex syntax.

I am more than happy to be shown otherwise I was unable to work it out sorry.

Reply:
>So with the github workflows the on command will look at a branch or a tag but it works like an or statement as far as I can tell. There is no way to and them.

Yep, I was thinking of them as an _and_. That's where I was going wrong.

>It would be much easier with a v<x>.<y>.<z> system which we could use independent of the branch and provide a bit more structure to the regex syntax.

I'm pretty happy to add the explicit `v` to version tags going forward. At my last gig we started with that and then dropped the `v` for reasons (not my own) which I don't think were all that valid in hindsight.

Are you ok to update this PR for `v<x>.<y>.<z>` (retaining the `release` branch restriction)? Will require a minor update to the `README.md` as well.